File: app/src/main/java/com/example/weatherapp/utils/dataprovider/WeatherDataProvider.java
package com.example.weatherapp.utils.dataprovider;

import com.example.weatherapp.model.LocationWeatherData;
import com.example.weatherapp.utils.dataprovider.exceptions.EmptyRequestException;
import com.example.weatherapp.utils.dataprovider.exceptions.NoSuchLocationException;

public class WeatherDataProvider {

    public static LocationWeatherData getData(String location)
            throws EmptyRequestException, NoSuchLocationException {

        if (location == null || location.isEmpty()) {
            throw new EmptyRequestException();
        }

        if (location.equalsIgnoreCase("moscow")) {

            double temperature = -25;
            double humidity = 50;
            double pressure = 740;
            double speedOfWind = 1;

            return new LocationWeatherData(location, temperature, humidity, pressure, speedOfWind);

        }

        if (location.equalsIgnoreCase("london")) {

            double temperature = -15.7;
            double humidity = 80;
            double pressure = 765;
            double speedOfWind = 4;

            return new LocationWeatherData(location, temperature, humidity, pressure, speedOfWind);

        }

        throw new NoSuchLocationException();

    }

}

File: weatherapp/app/src/main/java/com/example/weatherapp/utils/dataprovider/WeatherDataProvider.kt
package com.example.weatherapp.utils.dataprovider

import com.example.weatherapp.model.LocationWeatherData
import com.example.weatherapp.utils.dataprovider.exceptions.EmptyRequestException
import com.example.weatherapp.utils.dataprovider.exceptions.NoSuchLocationException

class WeatherDataProvider {

    companion object {

        fun getData(location : String) : LocationWeatherData {

            if (location.isEmpty())
                throw EmptyRequestException()

            if (location.equals("moscow", true))
                return LocationWeatherData(location,
                    -25.0,
                    50.0,
                    740.0,
                    1.0
                )

            if (location.equals("london", true))
                return LocationWeatherData(location,
                    -15.7,
                    80.0,
                    765.0,
                    4.0
                )

            throw NoSuchLocationException();

        }

    }

}